allen-cell-animated/timelapse-colorizer
# feature: Visualize annotations (annotations pt. 4)

Problem
=======
Part 4 of 5(ish) for #484, annotation prototyping! 

This change adds onscreen visualization for currently labeled object IDs, and also gives a nicer visual treatment to the annotation feature.

*Estimated review size: medium, 30-40 minutes*

Solution
========
- Annotated cells are shown visually with a colored marker matching the label. When multiple labels are present, a layered marker will be used.
- Labels on a cell can be viewed via the hover tooltip info area.
- In edit mode, the cursor changes to a crosshair and the current label follows the mouse.
- The canvas viewport now has a purple glow and a popup whenever annotation edit mode is enabled.
- Adds an onscreen visibility toggle for the annotations.

## Type of change
* New feature (non-breaking change which adds functionality)

Steps to Verify:
----------------
1. Open the preview link: https://allen-cell-animated.github.io/timelapse-colorizer/pr-preview-internal/pr-503/
2. Open the annotation tab and enable annotation edit mode. You should see:
  i. The viewport get a glow
  ii. The viewport should read "Annotation editing in progress..." in the top left corner
  iii. The mouse cursor should change to a crosshair
  iv. The current label ("Label 1") should follow the mouse
3. Click on a cell.
  i. The cell should get a green marker.
  ii. Hovering the cell will show the current labels ("Label 1").
4. Click on the cell again. The marker should disappear.
5. Click a few more cells, then create another label and click those same cells again. You should see the currently selected label on top, and a layered marker if a cell has multiple labels.


Screenshots (optional):
-----------------------

**Video talk-through (🔊):**

https://github.com/user-attachments/assets/98c3bdd6-a48d-4490-8081-2a5517a1d8c8

https://github.com/user-attachments/assets/7132ffdb-401a-4696-925a-5cc683e29d8b

![image](https://github.com/user-attachments/assets/67ab48bf-710e-411f-ad33-1577f21917f2)



Keyfiles (delete if not relevant):
-----------------------
1. `CanvasWrapper.tsx`
2. `annotations.ts`
3. `CanvasHoverTooltip.tsx`
